### ISSUE TITLE
Port twisted.protocols.policies to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6097]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6097|
|Type|enhancement|
|Created|2012-10-11 15:41:06Z|
|Branch|https://github.com/twisted/twisted/tree/policies-py3-6097|

`twisted.protocols.policies` should run on Python 3 (it's required by `twisted.protocols.tls`).

<details><summary>Searchable metadata</summary>

```
trac-id__6097 6097
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_policies_py3_6097 branches/policies-py3-6097
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349970066000000 1349970066000000
changetime__1349980303000000 1349980303000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36089]) Branching to 'policies-py3-6097'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review, I think: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/policies-py3-6097 has started.

  1. I improved coverage on `ProtocolWrapper` and `WrappedFactory`, which are what we really care about. Coverage is still imperfect elsewhere.
  1. For the logging factory, I decided to keep it as native strings written to the file; since it writes a `repr()` of the traffic, that doesn't seem like much of a problem.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. Typo in `_getWrapper` docstring - _conncected_
  1. ... `stopConsuming`?  Uh, ok whatever...

Looks good otherwise, please merge after addressing the above.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [36096]) Merge policies-py3-6097.

Author: itamar
Review: exarkun
Fixes: [#6097](https://github.com/twisted/twisted/issues/6097)

Port twisted.protocols.policies to Python 3.
